File: src/main/java/com/beat/domain/performance/application/dto/BookingPerformanceDetailResponse.java
package com.beat.domain.performance.application.dto;

import java.util.List;

public record BookingPerformanceDetailResponse(
        Long performanceId,
        String performanceTitle,
        String performancePeriod,
        List<BookingPerformanceDetailSchedule> scheduleList,
        int ticketPrice,
        String genre,
        String posterImage,
        String performanceVenue,
        String performanceTeamName,
        String bankName,
        String accountNumber,
        String accountHolder
) {
    public static BookingPerformanceDetailResponse of(
            Long performanceId,
            String performanceTitle,
            String performancePeriod,
            List<BookingPerformanceDetailSchedule> scheduleList,
            int ticketPrice,
            String genre,
            String posterImage,
            String performanceVenue,
            String performanceTeamName,
            String bankName,
            String accountNumber,
            String accountHolder
    ) {
        return new BookingPerformanceDetailResponse(performanceId, performanceTitle, performancePeriod, scheduleList, ticketPrice, genre, posterImage, performanceVenue, performanceTeamName, bankName, accountNumber, accountHolder);
    }

}

File: apis/src/main/kotlin/com/beat/apis/performance/api/response/BookingPerformanceDetailResponse.kt
package com.beat.apis.performance.api.response

import com.beat.global.support.jackson.CdnImageUrl

data class BookingPerformanceDetailResponse(
    val performanceId: Long?,
    val performanceTitle: String?,
    val performancePeriod: String?,
    val scheduleList: List<BookingPerformanceDetailScheduleResponse>,
    val ticketPrice: Int,
    val genre: String?, @field:CdnImageUrl val posterImage: String?,
    val performanceVenue: String?,
    val performanceTeamName: String?,
    val bankName: String?,
    val accountNumber: String?,
    val accountHolder: String?,
)
